fastestimator/backend/save_model.py

```diff
@@ -37,19 +41,31 @@ def save_model(model: Union[tf.keras.Model, torch.nn.Module], save_dir: str, mod
     Args:
         model: A neural network instance to save.
         save_dir: Directory into which to write the `model` weights.
-        model_name: The name of the model (used for naming the weights file).
+        model_name: The name of the model (used for naming the weights file). If None, model.model_name will be used.
+        save_optimizer: Whether to save optimizer. If True, optimizer will be saved in a separate file at same folder.
 
     Raises:
         ValueError: If `model` is an unacceptable data type.
     """
+    assert hasattr(model, "fe_compiled") and model.fe_compiled, "model must be built by fe.build"
+    if model_name is None:
+        model_name = model.model_name
     save_dir = os.path.normpath(save_dir)
     os.makedirs(save_dir, exist_ok=True)
     if isinstance(model, tf.keras.Model):
         model_path = os.path.join(save_dir, "{}.h5".format(model_name))
         model.save_weights(model_path)
+        if save_optimizer:
+            assert model.current_optimizer, "optimizer does not exist"
+            optimizer_path = os.path.join(save_dir, "{}_opt.pkl".format(model_name))
+            with open(optimizer_path, 'wb') as f:
+                pickle.dump(model.current_optimizer.get_weights(), f)
     elif isinstance(model, torch.nn.Module):
         model_path = os.path.join(save_dir, "{}.pt".format(model_name))
         torch.save(model.state_dict(), model_path)
+        if save_optimizer:
+            assert model.current_optimizer, "optimizer does not exist"
+            optimizer_path = os.path.join(save_dir, "{}_opt.pt".format(model_name))
+            torch.save(model.current_optimizer.state_dict(), optimizer_path)
     else:
         raise ValueError("Unrecognized model instance {}".format(type(model)))
-    print("FastEstimator-ModelSaver: saved model to {}".format(model_path))
```

Comment:
why are we moving the print away from where the actual saving happens?

Reply:
1. to be consistent with our printing philosophy: let Trace handles a formatted printing `FastEsrtimator-Trace: `
2. having the stuff on backend it does not look good when using `SpotRecovery` Trace.  that trace will save&load lots of models&optimizers and the printing message is just crazy.

Reply:
Then why is LoadModel adding print statements? 

print("Loaded optimizer from {}".format(optimizer_path))

Reply:
got that removed too